src/scenarios/personalization/ProgressSet/ProgressSet.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useState, useRef, useEffect, Component } from 'react';
+import {
+  Button,
+  Container,
+  Divider,
+  Form,
+  Grid,
+  Header,
+  Label,
+  Segment,
+} from 'semantic-ui-react';
+import {
+  useScenarioHost,
+  VariableSearch,
+} from '@axinom/mosaic-fe-samples-host';
+import { getApolloClient } from '../../../apollo-client';
+import { SetProgressMutation } from './graphql-documents';
+
+export const ProgressSet: React.FC = () => {
+  const { activeProfile, logger } = useScenarioHost();
+  const [FrequencyOfProgressSaving, setFrequencyOfProgressSaving] =
+    useState<number>();
+  const [userAccessToken, setUserAccessToken] = useState<string>();
+  const [key, setKey] = useState<string>();
+  const [buttonState, setButtonState] = useState<boolean>(true);
+  const [time, setTime] = useState(0);
+  const [intervalId, setIntervalId] = useState<NodeJS.Timer>();
+
+  const apolloClient = getApolloClient(
+    new URL('graphql', activeProfile.personalizationServiceBaseURL).href,
+  );
+
+  useEffect(() => {
+    let interval: ReturnType<typeof setInterval> | undefined;
```

Comment:
Since the return type of `setInterval` is a number, would it be simpler/cleaner to use `number | undefined` here?